realm/realm-library/src/main/java/io/realm/CollectionUtils.java

```diff
@@ -26,22 +26,43 @@
 import io.realm.internal.Table;
 import io.realm.internal.Util;
 
+
 public class CollectionUtils {
+    public static final String LIST_TYPE = "list";
+    public static final String DICTIONARY_TYPE = "dictionary";
+    public static final String SET_TYPE = "set";
 
     static boolean isClassForRealmModel(Class<?> clazz) {
         return RealmModel.class.isAssignableFrom(clazz);
     }
 
+    /**
+     * Called by both list and set operators to determine whether a RealmModel is an embedded object
+     *
+     * @param realm  the Realm instance to check against.
+     * @param object the object to check.
+     * @return true if the object can be copied, false otherwise
+     */
+    static boolean isEmbedded(BaseRealm realm, RealmModel object) {
```

Comment:
If we knew that compilation would fail for sets of embedded objects (see other comment) I guess we could eliminate this and all the runtime checks.